From f7f057bab90ea634b3e5c9c127c8b1c6151368f8 Mon Sep 17 00:00:00 2001
From: Willy Tarreau <w@1wt.eu>
Date: Thu, 24 Jan 2013 00:31:24 +0100
Subject: BUG/MINOR: config: check the proper variable when parsing log minlvl

logsrv->minlvl gets the numeric log level from the equivalent string.
Upon error, ->level was checked due to a wrong copy-paste. The effect
is that a wrong name will silently be ignored and due to minlvl=-1,
will act as if the option was not set.

No backport is needed, this is 1.5-specific.

Reported-by: Dinko Korunic <dkorunic@reflected.net>
---
 src/cfgparse.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/cfgparse.c b/src/cfgparse.c
index 718664c..9491147 100644
--- a/src/cfgparse.c
+++ b/src/cfgparse.c
@@ -4785,7 +4785,7 @@ stats_error_parsing:
 			logsrv->minlvl = 0; /* limit syslog level to this level (emerg) */
 			if (*(args[4])) {
 				logsrv->minlvl = get_log_level(args[4]);
-				if (logsrv->level < 0) {
+				if (logsrv->minlvl < 0) {
 					Alert("parsing [%s:%d] : unknown optional minimum log level '%s'\n", file, linenum, args[4]);
 					err_code |= ERR_ALERT | ERR_FATAL;
 					goto out;
-- 
1.7.1

